packages/kbn-alerts-ui-shared/src/rule_form/rule_definition/rule_alert_delay.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback } from 'react';
+import { EuiFieldText, EuiFormRow, EuiIconTip } from '@elastic/eui';
+import type { SanitizedRule, RuleTypeParams } from '@kbn/alerting-types';
+import {
+  ALERT_DELAY_TITLE_PREFIX,
+  ALERT_DELAY_TITLE_SUFFIX,
+  ALERT_DELAY_HELP_TEXT,
+  ALERT_DELAY_TITLE,
+} from '../translations';
+import { RuleFormErrors } from '../types';
+
+const INTEGER_REGEX = /^[1-9][0-9]*$/;
+
+export interface RuleAlertDelayProps {
+  alertDelay?: SanitizedRule<RuleTypeParams>['alertDelay'];
+  errors?: RuleFormErrors;
+  onChange: (property: string, value: unknown) => void;
```

Comment:
no because it could be an object or number or `null` to clear fields, etc. 

But this `onChange` is removed completed in favor of dispatches, which are typed correctly.